docs/rules/prop-types.md

```diff
@@ -91,6 +91,20 @@ class HelloEs6WithPublicClassField extends React.Component {
 }
 ```
 
+In TypeScript:
+
+```tsx
+// destructured default prop values
+
+function Foo({ bar = "" }): JSX.Element {
+  return <div>{bar}</div>;
+}
+
+function Foo({ bar = "" as string }): JSX.Element {
+  return <div>{bar}</div>;
```

Comment:
i'm a bit confused - are you saying that the props types would be inferred based on the types of the default values? what happens if there's even one required (non-defaulted) prop?